### ISSUE TITLE
regarding training codes

### ISSUE THREAD

**Issue by tao-bai** [Muzammal-Naseer/NRP#2]
Thanks for your sharing first! I notice that the training part is missing in the repo. Could you please update it or share a copy? Thank you so much!

**Comment by HuangZuShu** [comment]
It would be better, If @Muzammal-Naseer share the code and adversarial examples of this paper "Local Gradients Smoothing: Defense against localized adversarial attacks" published on WACV in 2019.

**Comment by Muzammal-Naseer** [comment]
> Thanks for your sharing first! I notice that the training part is missing in the repo. Could you please update it or share a copy? Thank you so much!

Hi Tao, NRP training is based on the following framework: https://github.com/xinntao/BasicSR
